### PR TITLE
Chilled beam airloop flow balance - maybe bigger than chilled beams

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
Attempting to address #6574 .
This may also address #6492 (see [comment here](https://github.com/NREL/EnergyPlus/issues/6574#issuecomment-407796677))

@EnergyArchmage Noted that `MassFlowRateZSup` is zero [here](https://github.com/NREL/EnergyPlus/blob/40101eaafd1525ee63406264ae9535f25acf4805/src/EnergyPlus/ZoneEquipmentManager.cc#L4487) in `CalcZoneReturnFlows`.

`MassFlowRateZSup` gets set [here](https://github.com/NREL/EnergyPlus/blob/40101eaafd1525ee63406264ae9535f25acf4805/src/EnergyPlus/ZoneAirLoopEquipmentManager.cc#L714-L742) in `SimZoneAirLoopEquipment`, but this code was never executing for this chilled beam (and for any other chilled beam for that matter), because `InNodeNum` = 0.

`InNodeNum` comes from `AirDistUnit( AirDistUnitNum ).InletNodeNum` which was never initialized for both flavors of chilled beams. That's been fixed here, but the system flow is still zero.

From what I see in the debugger at this point is that the airloop `MassFlowRateSetPoint` keeps ratcheting down the flow until the system ends up at zero. Wondering about the logic [here](https://github.com/NREL/EnergyPlus/blob/40101eaafd1525ee63406264ae9535f25acf4805/src/EnergyPlus/SimAirServingZones.cc#L2141-L2143) in `InitAirLoops` - but that code has been around a long time.

Making the exhaust fan "balanced" results in system flow, but that shouldn't be necessary. Need to investigate this more.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by nrel-bot-2** [comment]
@mjwitte @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mjwitte @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@mjwitte @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@mjwitte @lgentile it has been 34 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mjwitte @lgentile it has been 16 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@mjwitte @lgentile it has been 14 days since this pull request was last updated.

**Comment by rraustad** [comment]
Fix discussed via email corrects #6492.

image

**Comment by rraustad** [comment]
I added my code changes here so others could review. I did not address conflicting files at this time.

**Comment by rraustad** [comment]
The problem with this change is that the mixed air mass flow rate is artificially increased to account for the missing exhaust air (and happens to be the flow rate that the TUs want to operate at). That missing air mass flow is magically added to the recirc air stream when it should actually come from OA. So my thought is to add the flow discrepancy on top of the OA amount to maintain neutral building pressure. As @lgu1234 says, what happens if there is no outdoor air system? Or what if there is an OA system and the user has made up the exhaust with infiltration or ventilation? Still need to think through all of this.

**Comment by Myoldmopar** [comment]
@rraustad or @mjwitte please merge develop into this branch before you make any more pushes.  I am getting branches updated to use the new Ubuntu 18.04 CI specs, but I didn't take this one on because it has conflicts.

**Comment by nrel-bot-2** [comment]
@mjwitte it has been 7 days since this pull request was last updated.

**Comment by rraustad** [comment]
Very strange. I reverted all changes and kept just the one line of code that seemed most likely as the fix and results look reasonable.

fixzoneexhaust

Then I merged develop and got this.

fixzoneexhaust_mergeddevelop

File: src/EnergyPlus/InternalHeatGains.cc [hunk 1]
```diff
@@ -6116,7 +6116,7 @@ namespace InternalHeatGains {
         Real64 OperSchedFrac;                             // Operating schedule fraction
         Real64 CPULoadSchedFrac;                          // CPU loading schedule fraction
         Real64 AirConnection;                             // Air connection type
-        Real64 TSupply;                                   // Supply air temperature [C]
+        Real64 TSupply(0.0);                              // Supply air temperature [C]
```
Comment (rraustad) [inline src/EnergyPlus/InternalHeatGains.cc:6119]: Part of hack for #7041

File: src/EnergyPlus/InternalHeatGains.cc [hunk 2]
```diff
@@ -6313,7 +6313,7 @@ namespace InternalHeatGains {
                 TAirOut = TAirIn;
             }
 
-            if (abs(TAirOut - TSupply) < SmallTempDiff) {
+            if (SupplyNodeNum != 0 && abs(TAirOut - TSupply) < SmallTempDiff) {
```
Comment (rraustad) [inline src/EnergyPlus/InternalHeatGains.cc:6316]: Rest of hack for #7041

File: src/EnergyPlus/MixedAir.cc [hunk 3]
```diff
@@ -3758,8 +3758,8 @@ namespace MixedAir {
 
         } // if (AirLoopNum > 0)
 
-        // Set the relief air flow rate (must be done last to account for changes in OAMassFlow
-        this->RelMassFlow = max(this->OAMassFlow - this->ExhMassFlow, 0.0);
+          // Set the relief air flow rate (must be done last to account for changes in OAMassFlow
+        this->RelMassFlow = this->OAMassFlow - this->ExhMassFlow;
```
Comment (rraustad) [inline src/EnergyPlus/MixedAir.cc:3762]: Big question here is if relief air can be negative. Not sure where this limitation came from. Seems reasonable to expect no intake into relief node but this issue raises questions.

File: src/EnergyPlus/SimAirServingZones.cc [hunk 4]
```diff
@@ -2281,7 +2281,7 @@ namespace SimAirServingZones {
                 // Need to make sure that flows are greater than zero
                 if (MassFlowSet >= 0.0) {
                     Node(NodeNumOut).MassFlowRateSetPoint = MassFlowSet;
-                } else if (MassFlowSet < 0.0) {
+                } else {
```
Comment (rraustad) [inline src/EnergyPlus/SimAirServingZones.cc:2284]: excess logic, if mass flow rate is not >= it must be <

**Comment by rraustad** [comment]
@mjwitte passing this over to you. Enjoy.

**Comment by nrel-bot-3** [comment]
@mjwitte @lgentile it has been 14 days since this pull request was last updated.

**Comment by mjwitte** [comment]
Replaced with #7059

**Pull request closed without merging**